### DISCUSSION TITLE
Export the proof to a json file

### DISCUSSION THREAD

**Discussion by jeljelih** [Consensys/gnark#872] [Q&A]
I would like to use gnark to verify the proof by a solidity smart contract. Is there a way to export the proof as a json file? If not which other format do you recommend to export the proof and verify it from the solidity smart contract? Thank you in advance

**Comment by ivokub** [comment]
I wrote an example some time ago for verifying proof in smart contract. See https://github.com/ivokub/smartcontract. It isn't fully up to date, but hopefully you can get a gist.

See https://github.com/ivokub/smartcontract/blob/main/main.go#L246-L256 how to get the inputs necessary use as calldata.

Please keep in mind that for Groth16 the smart contract verifier is not fully compatible with native Go prover when using some features (field emulation etc.). It is WIP, but not a big priority right now.
